### PR TITLE
docs: table of contents for Queries and API pages

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary
Adds table of contents for Queries and APIs pages since they are quite long and hence hard to navigate.

### Test plan
None

### CODE REVIEW

File: website/docs/API.md [hunk 1]
```diff
@@ -519,40 +559,34 @@ A `RenderHookOptions<Props>` object to modify the execution of the `callback` fu
...
-```ts
-function rerender(newProps?: Props): void;
-```
-
-A function to rerender the test component, causing any hooks to be recalculated. If `newProps` are passed, they will replace the `callback` function's `initialProps` for subsequent rerenders.  The `Props` type is determined by the type passed to or inferred by the `renderHook` call.
+A function to rerender the test component, causing any hooks to be recalculated. If `newProps` are passed, they will replace the `callback` function's `initialProps` for subsequent rerenders. The `Props` type is determined by the type passed to or inferred by the `renderHook` call.
 
 #### `unmount`
 
-```ts
```
Comment (AugustinLF) [inline website/docs/API.md:552]: Why remove the API here while keeping it for `fireEvent.scroll: (element: ReactTestInstance, ...data: Array<any>) => void` for instance?

Comment (mdjastrzebski) [reply website/docs/API.md:552]: This `rerender` refers to `RenderHookResult` type which is kinda its own section. So I've put the whole type definition just above sections for first of it (`result`). Do you think we should repeat it here?

Comment (AugustinLF) [reply website/docs/API.md:552]: No you're right, it makes sense!

**Review by AugustinLF — APPROVED** [review]

**Review by thymikee — APPROVED** [review]
👍🏼

**Pull request merged**